src/mpid/ch4/shm/xpmem/xpmem_pre.h

```diff
@@ -21,19 +21,36 @@ typedef struct {
 } MPIDI_XPMEM_segmap_t;
 
 typedef struct MPIDI_XPMEM_seg {
-    off_t offset;               /* page aligned offset */
-    size_t size;                /* page aligned size */
-    void *vaddr;
+    /* AVL-tree internal components start */
+    struct MPIDI_XPMEM_seg *parent;
+    struct MPIDI_XPMEM_seg *left;
+    struct MPIDI_XPMEM_seg *right;
+    uint64_t min;               /* mininal address in this subtree */
+    uint64_t max;               /* maximal address in this subtree */
+    uint64_t height;            /* height of this subtree */
+    /* AVL-tree internal components end */
+
+    uint64_t low;               /* page aligned low address of remote seg */
+    uint64_t high;              /* page aligned high address of remote seg */
+    void *vaddr;                /* virtual address attached in current process */
+    int refcount;               /* reference count of this seg */
     MPIDI_XPMEM_segmap_t *segmap_ptr;   /* pointer to my segmap */
 } MPIDI_XPMEM_seg_t;
 
+
+typedef struct MPIDI_XPMEM_avl_tree {
+    MPIDI_XPMEM_seg_t *root;
+    int tree_size;
+} MPIDI_XPMEM_avl_tree_t;
+
 typedef struct {
     int num_local;
     int local_rank;
     MPIR_Group *node_group_ptr; /* cache node group, used at win_create. */
     xpmem_segid_t segid;        /* my local segid associated with entire address space */
     MPIDI_XPMEM_segmap_t *segmaps;      /* store registered segid for all local processes. */
     size_t sys_page_sz;
+    MPIDI_XPMEM_avl_tree_t *treemaps;   /* store AVL trees for all local processes */
```

Comment:
Can you please move `treemaps` into the front of `sys_page_sz` so that it looks better aligned ? It is just code-style preference.